packages/api/endpoints/executions.js

```diff
@@ -79,10 +81,35 @@ async function del(req, res) {
     throw error;
   }
 
-  await knex.transaction(async (trx) => {
-    await executionPgModel.delete(trx, { arn });
-    await executionModel.delete({ arn });
-  });
+  let apiExecution;
+  try {
+    apiExecution = await executionModel.get({ arn });
+  } catch (error) {
+    if (error instanceof RecordDoesNotExist) {
+      return res.boom.notFound('No record found');
+    }
+    throw error;
+  }
+
+  try {
+    await knex.transaction(async (trx) => {
+      await executionPgModel.delete(trx, { arn });
+      await executionModel.delete({ arn });
+      await deleteExecution({
+        esClient,
+        arn,
+        index: process.env.ES_INDEX,
+        ignore: [404],
+      });
+    });
+  } catch (error) {
+    // Delete is idempotent, so there may not be a DynamoDB
+    // record to recreate
+    if (apiExecution) {
+      await executionModel.create(apiExecution);
+    }
```

Comment:
Why do we need to recreate the dynamodb record? Wouldn't the execution model throw an error on line 86 if the record doesn't exist?

Reply:
The need to recreate the DynamoDB record comes from the transactional delete logic, which could happen this way:

- PG delete succeeds
- DynamoDB delete succeeds
- ES delete throws an error

In that case, the PG delete will get rolled back, but you can't roll back a DynamoDB delete. So the only way to "restore" the Dynamo record and thus have all three systems in the original state where the records exist is to manually recreate the DynamoDB record.